tests/scenario_certification/test_perturbation_family_registry.py

```
@@ -0,0 +1,156 @@
+
+
+def test_perturbation_family_unknown_raises() -> None:
+    """Unknown families should fail for any caller with a clear message."""
+    with pytest.raises(ValueError, match="unsupported perturbation family"):
+        perturbation_family("not_a_family")
+
+
+def test_perturbation_family_is_immutable() -> None:
+    """PerturbationFamily must be frozen for registry safety."""
+    family = perturbation_family("robot_route_offset")
+    with pytest.raises(Exception):
```

Ruff (0.15.14)

[warning] Do not assert blind exception: `Exception`

(B017)